src/transaction.rs

```diff
@@ -42,7 +46,9 @@ lazy_static! {
         LocalTxProver::from_bytes(&sapling_spend_bytes, &sapling_output_bytes)
     });
 }
-
+lazy_static! {
+    static ref TX_PROGRESS_LOCK: Mutex<u32> = Mutex::new(0);
```

Comment:
If we're keeping this a number, use an atomic int instead

Reply:
do we have atomic for floats?